### PR TITLE
[KIP-848] Added new configs group.protocol and group.remote.assignor

### PR DESCRIPTION
**Pull request by pranavrth**

### CODE REVIEW

File: src/rdkafka_cgrp.c [hunk 1]
```diff
@@ -420,6 +422,8 @@ rd_kafka_cgrp_t *rd_kafka_cgrp_new(rd_kafka_t *rk,
         rkcg->rkcg_q                        = rd_kafka_consume_q_new(rk);
         rkcg->rkcg_group_instance_id =
             rd_kafkap_str_new(rk->rk_conf.group_instance_id, -1);
+        rkcg->rkcg_group_remote_assignor =
```
Comment (emasab) [inline src/rdkafka_cgrp.c:425]: Add destruction of this field in `rd_kafka_cgrp_destroy_final`

File: src/rdkafka_conf.c [hunk 2]
```diff
@@ -1135,6 +1135,21 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
...
+    {_RK_GLOBAL | _RK_CGRP | _RK_HIGH, "group.protocol", _RK_C_S2I,
+     _RK(group_protocol),
+     "Group protocol to use. Use `generic` for the old protocol "
+     "and `consumer` for the new protocol introduced in KIP-848. "
+     "Availabe protocols: generic or consumer. Default is `generic` but "
+     "is expected to be changed to `consumer` in future.",
+     .vdef = RD_KAFKA_GROUP_PROTOCOL_GENERIC,
+     .s2i  = {{RD_KAFKA_GROUP_PROTOCOL_GENERIC, "generic"},
+             {RD_KAFKA_GROUP_PROTOCOL_CONSUMER, "consumer"}}},
+    {_RK_GLOBAL | _RK_CGRP | _RK_MED, "group.remote.assignor", _RK_C_STR,
```
Comment (emasab) [inline src/rdkafka_conf.c:1147]: We could hide this with `_RK_HIDDEN` as it won't be available in the preview.

File: CONFIGURATION.md [hunk 3]
```diff
@@ -110,6 +110,8 @@ partition.assignment.strategy            |  C  |                 | range,roundro
...
 heartbeat.interval.ms                    |  C  | 1 .. 3600000    |          3000 | low        | Group session keepalive heartbeat interval. <br>*Type: integer*
 group.protocol.type                      |  C  |                 |      consumer | low        | Group protocol type. NOTE: Currently, the only supported group protocol type is `consumer`. <br>*Type: string*
+group.protocol                           |  C  | generic, consumer |       generic | high       | Group protocol to use. Use `generic` for the old protocol and `consumer` for the new protocol introduced in KIP-848. Availabe protocols: generic or consumer. Default is `generic` but is expected to be changed to `consumer` in future. <br>*Type: enum value*
```
Comment (emasab) [inline CONFIGURATION.md:113]:
```suggestion
group.protocol                           |  C  | generic, consumer |       generic | high       | Group protocol to use. Use `generic` for the original protocol and `consumer` for the new protocol introduced in KIP-848. Available protocols: generic or consumer. Default is `generic`, but will change to `consumer` in next releases. <br>*Type: enum value*
```

Comment (emasab) [reply CONFIGURATION.md:113]: Change to `Group protocol type for the generic protocol` in `group.protocol.type` description.

**Review by emasab — COMMENTED** [review]
Just a few comments and LGTM

File: CONFIGURATION.md (hunk 3 above)
Comment (pranavrth) [reply CONFIGURATION.md:113]: I am not touching `group.protocol.type`. It is an old property with different use.

Comment (emasab) [reply CONFIGURATION.md]: Only the documentation, as it's for the generic protocol, otherwise it's confusing

File: src/rdkafka_conf.c [hunk 4]
```diff
@@ -1135,6 +1135,21 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
      "Group protocol type. NOTE: Currently, the only supported group "
      "protocol type is `consumer`.",
      .sdef = "consumer"},
+    {_RK_GLOBAL | _RK_CGRP | _RK_HIGH, "group.protocol", _RK_C_S2I,
+     _RK(group_protocol),
+     "Group protocol to use. Use `generic` for the old protocol "
```
Comment (emasab) [inline src/rdkafka_conf.c:1140]: Apply the documentation changes here and run `make CONFIGURATION.md` to regenerate the file.

File: src/rdkafka_conf.c [hunk 5]
```diff
@@ -1132,9 +1132,26 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
      "Group session keepalive heartbeat interval.", 1, 3600 * 1000, 3 * 1000},
     {_RK_GLOBAL | _RK_CGRP, "group.protocol.type", _RK_C_KSTR,
      _RK(group_protocol_type),
-     "Group protocol type. NOTE: Currently, the only supported group "
+     "Group protocol type for the `generic` group protocol. NOTE: Currently, "
+     "the only supported group "
      "protocol type is `consumer`.",
      .sdef = "consumer"},
+    {_RK_GLOBAL | _RK_CGRP | _RK_HIGH | _RK_HIDDEN, "group.protocol", _RK_C_S2I,
```
Comment (emasab) [inline src/rdkafka_conf.c:1139]: Do we leave `group.protocol` out of the documentation for the preview? It's excluded because of the `_RK_HIDDEN`

Comment (pranavrth) [reply src/rdkafka_conf.c:1139]: I think we should hide it for now. If required, we will update it.

**Review by emasab — APPROVED** [review]

**Pull request merged**